### ISSUE TITLE
Update msftidy to detect incorrectly specified Rank

### ISSUE THREAD

**Issue by jhart-r7** [rapid7/metasploit-framework#6366]
Today, auxiliary modules can have a Rank specified but this does nothing and just results in copy-pasta.  If this is not needed, we should enforce it with msftidy.

**Comment by wchen-r7** [comment]
Ranks are definitely supported for aux. How would you properly detect the correct rank for a module?

**Comment by jhart-r7** [comment]
Right now, no auxiliary modules appear to specify Rank, so perhaps I jumped to an incorrect conclusion here.  If it is supported and it makes sense for auxiliary modules, then we can close this.

**Comment by wchen-r7** [comment]
You are correct no auxiliary modules actually specify Rank, but since all modules default to NormalRanking when nothing is specified, plus you can see it on the console, that makes it look like we support Rank in auxiliary. However, I don't think we have ever defined rankings for auxiliary modules either, only for exploits: https://github.com/rapid7/metasploit-framework/wiki/Exploit-Ranking.... so I guess my question is, if we do support aux, maybe we need to start defining the rankings? Or if we don't care, don't set a default and display it on the console in the first place?

Up for discussion :-)

**Comment by wvu-r7** [comment]
This feels familiar, @jhart-r7... maybe #3158 is why you're not seeing aux modules with rank?

**Comment by jhart-r7** [comment]
That seems to be related but also not.  The original intent with #3158 looks to be similar to what I'm asking for here -- to have msftidy warn or error if auxiliary modules use Rank because they make a little less sense there (https://github.com/rapid7/metasploit-framework/commit/d9df2fbf08037f409ca174b05e5c7982c98d2067).  But, what was actually merged in https://github.com/rapid7/metasploit-framework/commit/481d72bbe3ed01c91dd84d61196fe7da92822be6 (also seen in https://github.com/rapid7/metasploit-framework/pull/3158/files) simply removed the one Ranking that existed in all of the auxiliary modules at the time.

**Comment by jhart-r7** [comment]
After discussions with @wvu-r7 about this, it appears what we are seeing as the final result in #3158 according to github is misleading due to a branch name being re-used at one point which confuses the history and final result as shown in github.

As currently committed, `msftidy` will warn when a `Rank` is found in auxiliary modules (I modified the module below to specify a Rank.

```
modules/auxiliary/dos/http/apache_commons_fileupload_dos.rb:9 - [WARNING] Auxiliary modules have no 'Rank': Rank = LowRanking 
```

I'd like to see this move to `error`, but my only real argument for that is that it would help prevent copy-pasta.  Closing.

Thanks for clearing things up, @wvu-r7
